test/node/error_test.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const BSON = require('../register-bson');
+const BSONTypeError = BSON.BSONTypeError;
+const BSONError = BSON.BSONError;
+
+describe('BSON error tests', function () {
+  it('should correctly validate instanceof checks for BSONTypeError and BSONError', function () {
+    const bsonTypeErr = new BSONTypeError();
+    const bsonErr = new BSONError();
+
+    expect(bsonTypeErr instanceof BSONTypeError).to.be.true;
+    expect(bsonErr instanceof BSONError).to.be.true;
```

Comment:
We can use chai's assert here for some more readability and also just add a check that it still correctly walks the protoype chain to return true for the base class:

```js
expect(bsonTypeErr).to.be.instanceOf(BSONTypeError);
expect(bsonTypeErr).to.be.instanceOf(TypeError);
```